### DISCUSSION TITLE
How do you deal with extra data in GetCollection?

### DISCUSSION THREAD

**Discussion by lyrixx** [api-platform/core#5660] [Q&A]
I have an endpoint `GET /articles`, And I want to ship more data than just the
articles. For example, I want to ship some aggregations, the max price, the
lowest price, etc. Example (see the last bits):

```json
{
  "@context": "\/api\/v3\/contexts\/Article",
  "@id": "\/api\/v3\/brand\/articles",
  "@type": "hydra:Collection",
  "hydra:totalItems": 1000,
  "hydra:member": [
    {
      "@id": "\/api\/v3\/brand\/articles\/101",
      "@type": "Article",
      "id": 101,
      "condition": "FIXME",
      "color": "FIXME",
      "size": "FIXME"
    },
    {
      "@id": "\/api\/v3\/brand\/articles\/102",
      "@type": "Article",
      "id": 102,
      "condition": "FIXME",
      "color": "FIXME",
      "size": "FIXME"
    },
    # ...
  ],
  "hydra:view": {
    "@id": "\/api\/v3\/brand\/articles?page=1",
    "@type": "hydra:PartialCollectionView",
    "hydra:first": "\/api\/v3\/brand\/articles?page=1",
    "hydra:last": "\/api\/v3\/brand\/articles?page=100",
    "hydra:next": "\/api\/v3\/brand\/articles?page=2"
  },
  "aggregations": {
    "by_price": {...},
    "by_color": {...},
  },
  "stats": {
    "max_price": 10,
    "min_price": 1,
  }
}
```

I used to do this with a data provider, a DTO, and a Normalizer. But now I want
to expose also the type of the extra data (aggregations, stats, etc) to the
consumer of the API (auto generation of the the typescript client).

And I would like to avoid to fill manually all the types in an extension of the
Swagger Normalizer.

Have you already faced this problem? How did you solve it?

**Comment by lyrixx** [comment]
For the record, I build a system hook in the Open Api Factory, to add the type (aggregations) I need.

If some people are interested, I can write a blog post about it

**Reply by divine** [reply]
Hello,

I'm interested in a blog post on how you achieved that!

Thanks!
